catalog/app/components/FileEditor/QuiltConfigEditor/BucketPreferences/PackageDescription.tsx

```diff
@@ -0,0 +1,277 @@
+import * as M from '@material-ui/core'
+import * as Lab from '@material-ui/lab'
+
+import type { PackagePreferencesInput } from 'utils/BucketPreferences/BucketPreferences'
+
+import type { KeyedValue } from './State'
+
+interface JsonPathsProps {
+  disabled?: boolean
+  onChange: (value: string[]) => void
+  size: 'small' | 'medium'
+  value?: readonly string[]
+}
+
+const options: string[] = []
+
+const emptyArray: string[] = []
+
+function JsonPaths({ disabled, onChange, size, value = [] }: JsonPathsProps) {
+  const [error, setError] = React.useState<Error | null>(null)
+  const handleChange = React.useCallback(
+    (_e, labels: string[]) => {
+      onChange(labels)
+      try {
+        labels.forEach((label: string) => jsonpath.parse(label))
+        setError(null)
+      } catch (err) {
+        setError(err instanceof Error ? err : new Error('Invalid JSON path'))
+      }
+    },
+    [onChange],
+  )
+  return (
+    <Lab.Autocomplete
+      disabled={disabled}
+      freeSolo
+      multiple
+      options={options}
+      onChange={handleChange}
+      renderInput={(params) => (
+        <M.TextField
+          {...params}
+          InputLabelProps={{
+            shrink: true,
+          }}
+          error={!!error}
+          helperText={error?.message}
+          label="JSON paths from `user_meta`"
+          placeholder="$.Some.Key, $.Another.Key"
+        />
+      )}
+      size={size}
+      value={(value as string[]) || emptyArray}
+    />
+  )
+}
+
+interface MessageProps {
+  disabled?: boolean
+  onChange: (value: boolean) => void
+  size: 'small' | 'medium'
+  value?: boolean
+}
+
+function Message({ disabled, onChange, size, value = false }: MessageProps) {
+  const handleChange = React.useCallback(
+    (_e, checked: boolean) => onChange(checked),
+    [onChange],
+  )
+  return (
+    <M.FormControl>
+      <M.FormControlLabel
+        control={<M.Checkbox checked={value} size={size} onChange={handleChange} />}
+        disabled={disabled}
+        label="Display commit message"
+      />
+    </M.FormControl>
+  )
+}
+
+interface PackageHandleProps {
+  disabled?: boolean
+  onChange: (value: string) => void
+  size: 'small' | 'medium'
+  value: string
+}
+
+function PackageHandle({
+  disabled,
+  onChange,
+  size,
+  value: initialValue,
+}: PackageHandleProps) {
+  const [value, setValue] = React.useState(initialValue)
+  const handleEnter = React.useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>) => {
+      if (event.key === 'Enter') {
+        onChange(value)
+      }
+    },
+    [onChange, value],
+  )
+  const handleChange = React.useCallback((event) => setValue(event.target.value), [])
+  return (
+    <M.TextField
+      InputLabelProps={{
+        shrink: true,
+      }}
+      disabled={disabled}
+      label="RegExp or name for the package handle"
+      onBlur={() => onChange(value)}
+      onChange={handleChange}
+      onKeyDown={handleEnter}
+      placeholder="namespace/production"
```

Comment:
```suggestion
      placeholder="e.g. ^prefix/.*"
```
looking at the code, i see that regexp matching is used in every case, there's no literal matching, so `namespace/production` will match every package name containing that substring, not only that specific package. for literal matching, a pattern must be anchored with `^` and `$`.
we should fix the docs btw, because the example there is incorrect